backend/src/src-vulnerabilities/src/main/java/org/eclipse/sw360/vulnerabilities/VulnerabilityHandler.java

```diff
@@ -202,17 +196,38 @@ private void enrichVulnerabilityDTO(VulnerabilityDTO dto, String releaseId, User
                     releaseName = release.getName() + " ";
                     dto.setIntComponentName(release.getName());
                 } else {
-                    Component component = compHandler.getComponent(release.getComponentId(), user);
+                    Component component = componentCache.get(release.getComponentId());
                     if (component != null) {
                         releaseName = component.getName() + " ";
                         dto.setIntComponentName(component.getName());
                     }
                 }
                 dto.setIntReleaseName(releaseName + release.getVersion());
             }
-        } catch (SW360Exception e) {
+        } catch (ExecutionException e) {
             log.error(e);
         }
+        return dto;
+    }
+
+    private LoadingCache<String, Release> makeReleaseCache(User user) {
+        return CacheBuilder.newBuilder()
+                .build(new CacheLoader<String, Release>() {
+                    public Release load(String key) throws SW360Exception {
```

Comment:
This will still make pretty many roundtrips to the DB. Did you test the performance change? Does the caching bring much? Did you think of loading all necessary releases in one go (if that's possible at all)?

The same applies, obviously, to components cache as well.

Reply:
The number of roundtrips to the DB is much reduced. In the past, each VulnerabilityReleaseRelation was triggered one call to fetch its associated release. In practice, releases typically have multiple vulnerabilities, which means the same release was loaded many times (typically 10-20 times, up to 200 for Linux Kernel and the like).
For the component cache, the component is loaded only if the release name is empty. So in practice there is nearly no usage of that cache.
I suppose it is possible to further refactor the code to load all releases at one, but this is quite some work.

Reply:
Sounds good